test/shmemx/shmemx_malloc_with_hints.c

```diff
@@ -0,0 +1,111 @@
+/*
+ *  Copyright (c) 2020 Intel Corporation. All rights reserved.
+ *  This software is available to you under the BSD license below:
+ *
+ *      Redistribution and use in source and binary forms, with or
+ *      without modification, are permitted provided that the following
+ *      conditions are met:
+ *
+ *      - Redistributions of source code must retain the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer.
+ *
+ *      - Redistributions in binary form must reproduce the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer in the documentation and/or other materials
+ *        provided with the distribution.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include <stdio.h>
+#include <stdint.h>
+#include <shmem.h>
+#include <shmemx.h>
+
+#define N 128
+#define MAX(a,b) ((a) > (b)) ? a : b
+#define WRK_SIZE MAX(N/2+1, SHMEM_REDUCE_MIN_WRKDATA_SIZE)
+
+#define NO_TESTS 5
+
+#define SHMEMX_MALLOC_INVALID_HINT 1l<<3
+
+long pSync[SHMEM_ALLTOALL_SYNC_SIZE];
+int pWrk[WRK_SIZE];
+
+
+int sumtoall_with_malloc_hint(long hint, int mype, int npes)
+{
+    int failed = 0;
+    int i;
+    int *dst, *src;
+
+    if (mype == 0)
+        printf("Sum reduction operation started\n");
+
+    dst = (int *)shmemx_malloc_with_hints(N * sizeof(*dst), hint);
+    src = (int *)shmemx_malloc_with_hints(N * sizeof(*src), hint);
+
+    for (i = 0; i < N; i++) {
+        src[i] = mype;
+        dst[i] = -1;
+    }
+
+    shmem_barrier_all();
+    shmem_int_sum_to_all(dst, src, N, 0, 0, npes, pWrk, pSync);
+
+    if(mype == 0){
+        for (i = 0; i < npes; i++) {
+            int expected;
+            expected = npes * (npes-1)/2;
+            if (dst[i] != expected) {
+                printf("[%d] dst[%d] = %d, expected %d\n", mype, i, dst[i], expected);
+                failed = 1;
+            }
```

Comment:
Sorry for late comments. But I have 2 here.
1. minor: if<space>(mype == 0). Similar comment to Dave's earlier one. I think there is another instance below.. 
2. Shouldn't we check for all the N values in the for loop?

Reply:
good points! 2nd point was overlook on my part. It should be checked for N values. I'll make the changes. Thanks!